modules/serverless-architecture.adoc

```diff
@@ -0,0 +1,35 @@
+// Module included in the following assemblies:
+
+// * serverless-overview.adoc
+
+[id="serverless-architecture_{context}"]
+= Red Hat OpenShift Serverless product architecture
+
+toc::[]
+
+== Knative Serving
+
+Knative Serving on OpenShift builds on Kubernetes and Maistra to support deploying and serving serverless applications.
+
+It creates a set of Kubernetes Custom Resource Definitions (CRDs) that are used to define and control the behavior of serverless workloads on an OpenShift cluster.
+
+These CRDs can be used as building blocks to address complex use cases, such as rapid deployment of serverless containers, automatic scaling of pods, routing and network programming for Maistra components, or viewing point-in-time snapshots of deployed code and configurations.
+
+=== Knative Serving components
+
+==== Knative service resource
+
+The `service.serving.knative.dev` resource automatically manages the whole lifecycle of a serverless workload on a cluster. It controls the creation of other objects to ensure that an app has a route, a configuration, and a new revision for each update of the service. Services can be defined to always route traffic to the latest revision or to a pinned revision.
+
+==== Knative route resource
+The `route.serving.knative.dev` resource maps a network endpoint to one or more Knative revisions. You can manage the traffic in several ways, including fractional traffic and named routes.
+
+==== Knative configuration resource
+The `configuration.serving.knative.dev` resource maintains the desired state for your deployment. Modifying a configuration creates a new revision.
+
+==== Knative revision resource
+The `revision.serving.knative.dev` resource is a point-in-time snapshot of the code and configuration for each modification made to the workload. Revisions are immutable objects and can be retained for as long as needed.
+Cluster administrators can modify the `revision.serving.knative.dev` resource to enable automatic scaling of pods in your OpenShift cluster.
+
+== Knative Client
+Knative Client (`kn`) extends the functionality of `oc` or `kubectl` to enable interaction with Knative components on OpenShift. `kn` allows developers to deploy and manage applications without editing YAML files directly.
```

Comment:
The Knative Client

I'd s/of `oc` or `kubectl`/ of the `oc` or `kubectl` tools  (https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/term_glossary.adoc#openshift-cli)